instrumentation/apache-httpclient/apache-httpclient-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v2_0/ApacheHttpClientSingletons.java

```diff
@@ -28,16 +29,17 @@
         HttpSpanNameExtractor.create(httpAttributesExtractor);
     SpanStatusExtractor<? super HttpMethod, ? super HttpMethod> spanStatusExtractor =
         HttpSpanStatusExtractor.create(httpAttributesExtractor);
-    ApacheHttpClientNetAttributesExtractor netAttributesExtractor =
-        new ApacheHttpClientNetAttributesExtractor();
-
+    ApacheHttpClientNetAttributesGetter netAttributesAdapter =
+        new ApacheHttpClientNetAttributesGetter();
+    NetClientAttributesExtractor<HttpMethod, HttpMethod> netAttributesExtractor =
+        NetClientAttributesExtractor.create(netAttributesAdapter);
```

Comment:
Nit: could this one be inlined?

Reply:
sure, it's just a massive wall of text anyway..... 🙃